### ISSUE TITLE
TimeoutException: com.sun.mail.imap.IMAPSSLStore.finalize()

### ISSUE THREAD

**Issue by grote** [konradrenner/kolabnotes-android#81]
This came a few times while I was trying to sync my account and made the app crash:

```
11-26 17:49:03.900 E/AndroidRuntime(20297): Process: org.kore.kolabnotes.android:sync, PID: 20297
11-26 17:49:03.900 E/AndroidRuntime(20297): java.util.concurrent.TimeoutException: com.sun.mail.imap.IMAPSSLStore.finalize() timed out after 10 seconds
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.android.org.conscrypt.OpenSSLSocketImpl$SSLInputStream.read(OpenSSLSocketImpl.java:699)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.util.TraceInputStream.read(TraceInputStream.java:124)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at java.io.InputStream.read(InputStream.java:162)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at java.io.BufferedInputStream.fillbuf(BufferedInputStream.java:149)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at java.io.BufferedInputStream.read(BufferedInputStream.java:234)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.iap.ResponseInputStream.readResponse(ResponseInputStream.java:96)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.iap.Response.<init>(Response.java:109)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.protocol.IMAPResponse.<init>(IMAPResponse.java:60)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.protocol.IMAPProtocol.readResponse(IMAPProtocol.java:321)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.iap.Protocol.command(Protocol.java:315)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.protocol.IMAPProtocol.logout(IMAPProtocol.java:382)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.IMAPFolder.close(IMAPFolder.java:1441)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.IMAPFolder.close(IMAPFolder.java:1384)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.IMAPStore.cleanup(IMAPStore.java:1649)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.IMAPStore.releaseStoreProtocol(IMAPStore.java:1276)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.IMAPStore.close(IMAPStore.java:1577)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at com.sun.mail.imap.IMAPStore.finalize(IMAPStore.java:1583)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:191)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:174)
11-26 17:49:03.900 E/AndroidRuntime(20297):     at java.lang.Thread.run(Thread.java:818)
```

**Comment by konradrenner** [comment]
I think if issue #82 is solved, this issue is also solved. So I will concetrate on issue #82

**Comment by konradrenner** [comment]
So that #82 is closed, do you still force this problem ?

**Comment by grote** [comment]
I haven't seen this anymore, but it might still happen with a bad network connection. I would catch this exception in any case and mark the sync as failed instead of letting the app crash.

**Comment by konradrenner** [comment]
Ok, this is a good point. I will do it this way ;-)

**Comment by konradrenner** [comment]
Should be fixed with 1.1.0
